### PR TITLE
fix multiple page calls issue

### PR DESCRIPTION
**Pull request by hankim813**
@f2prateek @sperand-io  this is for server side. 

I'm using the `sent = false` flag in this one because we can't use a return since we are using a batch to send the request. So instead, I'm just selectively pushing only once to the batch. 

Also had to fix the tests because they were -- just like client side-- incorrectly testing for the behavior in the first place.

### CODE REVIEW

**Comment by f2prateek** [comment]
We don't really make multiple calls anymore - so I'd get rid of the `batch` (and `sent`) entirely since we only need to make one call.

**Comment by hankim813** [comment]
@f2prateek got rid of the batch and just ported the message through the track mapper.

can you merge once you take a look?

File: lib/index.js [hunk 1]
```diff
@@ -182,37 +182,36 @@ Mixpanel.prototype.track = function(track, fn){
 
 Mixpanel.prototype.screen =
 Mixpanel.prototype.page = function(msg, fn){
-  var batch = new Batch;
   var category = msg.category();
-  var name = msg.fullName();
+  var name = msg.name();
+  var fullName = msg.fullName(); // includes both category and name
   var self = this;
+  var sent = false; // Prevents sending more than one call per page/screen
```
Comment (f2prateek) [inline lib/index.js]: We don't need this flag anymore. Just return after you make the first call to `send`.

**Comment by hankim813** [comment]
@f2prateek alright I think it's good to go!

**Comment by f2prateek** [comment]
LGTM!

File: lib/index.js [hunk 2]
```diff
@@ -182,37 +182,39 @@ Mixpanel.prototype.track = function(track, fn){
...
-    };
+  // call via track method with `msg`.
+  function send(msg, done){
+    self.track(msg, function(err, arr){
+      if (err) return done(err);
+      done(null, arr[1]);
+    });
   }
-
-  batch.end(fn);
```
Comment (calvinfo) [inline lib/index.js]: hey guys, we have to call a callback here or else it won't actually complete, and it'll end up hanging.

**Pull request merged**